### DISCUSSION TITLE
REFS vs REFERENCES of Threads algorithm

### DISCUSSION THREAD

**Discussion by bilogic** [the-djmaze/snappymail#1819] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/721d4924-9e77-436f-928d-e4ba702b4db7)

What's the difference between REFS vs REFERENCES?

**Comment by Potherca** [comment, marked as answer]
`REFS` is an improved version of `REFERENCE`. 

`REFERENCE` might group messages together that have the same Subject but are otherwise unrelated. `REFS` tries to avoid this.

<details><summary>Sources</summary>

The `THREAD` algorithm is part of [the SORT and THREAD Extensions to the Internet Message Access (IMAP) Protocol (RFC-5256)](https://datatracker.ietf.org/doc/html/rfc5256). That spec defines `ORDEREDSUBJECT` (sorted by subject and then by date) and `REFERENCES` (grouped based on which messages are replies to others).

`THREAD=REFS` is defined in a later spec, [The IMAP SEARCH=INTHREAD and THREAD=REFS Extensions](https://datatracker.ietf.org/doc/html/draft-ietf-morg-inthread), which never reached full RFC status.

To quote that spec:

> The THREAD=REFS thread algorithm is defined as the part of THREAD=REFERENCES (see [[RFC5256](https://datatracker.ietf.org/doc/html/rfc5256)]) which concerns itself with the References, In-Reply-To and Message-ID fields.  THREAD=REFS ignores Subject.
>
> THREAD=REFS sorts threads by the most recent INTERNALDATE in each thread, replacing THREAD=REFERENCES step (4). This means that when a new message arrives, its thread becomes the latest thread. (Note that while threads are sorted by arrival date, messages within a thread are sorted by sent date, just as for THREAD=REFERENCES.)
>
> This document defines THREAD=REFS because THREAD=REFERENCES is too inclusive for the INTHREAD search key. 
> For example, independent threads that happen to have the same subject field (such as "Agenda for Friday's meeting", "Web site updated" or "Message delivery failed") are grouped into one thread by THREAD=REFERENCES.

Compare that to RFC-5256:

> The REFERENCES threading algorithm threads the searched messages by grouping them together in parent/child relationships based on which messages are replies to others.
>
> The parent/child relationships are built using two methods: reconstructing a message's ancestry using the references contained within it; and checking the original (not base) subject of a message to see if it is a reply to (or forward of) another message.

</details>
